packages/shared/ReactScheduler.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+import {
+  now,
+  scheduleWork,
+  cancelScheduledWork,
+} from 'react-scheduler/src/ReactScheduler';
```

Comment:
This indirection seems a bit complicated :-(

Do we want to keep this fork long term? Is there a better fix in mind? How will this ultimately work in open source when the package is ready?

Reply:
It is complicated, agreed.

> Do we want to keep this fork long term? Is there a better fix in mind? How will this ultimately work in open source when the package is ready?

Here is my understanding of how it will work - 
- Once the `schedule` API is stable, we publish it as a separate module and make it a dependency of `react-dom` and `react-art`. In OS people just pull in the `schedule` bundle via npm, and it uses the native `requestAnimationFrame`.
- We would still need something like this for the 'www' build of 'react-dom', because we still have the problem with our 'requestAnimationFrame' polyfill in 'www'. Some of the behavior of the polyfill doesn't work, but some of the custom behavior we still need. I think at that point we can get away with the following simpler approach in 'www':
    - we still require 'schedule' before the polyfill runs, which means 'schedule' pulls in the native rAF and uses it.
    - In 'www' we write a wrapper module that adds the 'timeSlice' wrapper to 'schedule' itself, which we still need, and React would grab the wrapped version. Not sure if we'll still need a fork/shim for that to work.

It took a lot of discussion to agree on this solution, I'm hesitant to go back to the drawing board when we have something that works. 

Regarding my original solution of just shimming 'requestAnimationFrame' itself in our 'www' build - @sebmarkbage had concerns that we should not be using the polyfilled version of 'requestAnimationFrame' at all in 'www', and that the best way to avoid pulling in the polyfill was to require the `schedule` module before the polyfilling happens.

Reply:
Actually, if we just wrap the 'callback' before we pass it into 'scheduleWork' at every single callsite, or even if we just expose a wrapped version of 'schedule' for non-React use within 'www', then I think we can still avoid a fork of 'schedule'. After we publish it as a separate module, that is.